core/src/cluster_nodes.rs

```diff
@@ -313,6 +317,19 @@ impl ClusterNodes<RetransmitStage> {
     }
 }
 
+impl ClusterNodes<FindPacketSenderStakeStage> {
+    pub(crate) fn get_ip_to_stakes(&self) -> HashMap<IpAddr, u64> {
+        self.compat_index
+            .iter()
+            .filter_map(|(_, i)| {
+                let node = &self.nodes[*i];
+                let contact_info = node.contact_info()?;
+                Some((contact_info.tvu.ip(), node.stake))
+            })
+            .collect()
+    }
```

Comment:
seems to me this has to be cached on construction.
and then maybe wrap it in an `Arc` so that clones are cheap